src/main/java/net/coderbot/iris/Iris.java

```diff
@@ -174,6 +175,12 @@ private static void loadInternalShaderpack() {
 		logger.info("Using internal shaders");
 	}
 
+	private static void loadNoOpShaderPack() {
+		currentPack = ShaderPack.NO_OP;
```

Comment:
If there was a problem loading the `NO_OP` shaderpack, it get's set to null [here](https://github.com/FoundationGames/Iris/blob/disable_shaders/src/main/java/net/coderbot/iris/shaderpack/ShaderPack.java#L260). Currently if the currentPack is also null, it will just throw at a `requireNull` check [here](https://github.com/IrisShaders/Iris/blob/trunk/src/main/java/net/coderbot/iris/Iris.java#L233), so maybe check if currentPack is null and then fallback to internal if it is with a log error?

Reply:
Ideally, the no-op pack should never fail to load (see above), since it's hardcoded. One of the reasons to add the no-op shader pack is that it's a more reliable fallback than the internal shaders, and that it could become the new fallback (may be implemented in a future commit if this isn't a bad idea in some way).